QmlNet/QmlNet/qml/NetQModelIndex.cpp

```diff
@@ -4,19 +4,10 @@
 
 extern "C" {
 
-Q_DECL_EXPORT int net_qmodelindex_column(NetQModelIndexContainer* container) {
-    return container->modelIndex->column();
-}
-Q_DECL_EXPORT int net_qmodelindex_row(NetQModelIndexContainer* container) {
-    return container->modelIndex->row();
-}
-Q_DECL_EXPORT NetQModelIndexContainer* net_qmodelindex_parent(NetQModelIndexContainer* container) {
-    return new NetQModelIndexContainer{
-        .modelIndex = QSharedPointer<QModelIndex>(&container->modelIndex->parent())
-    };
-}
-Q_DECL_EXPORT void net_qmodelindex_destroy(NetQModelIndexContainer* container) {
-    delete container;
-}
+Deleter(QModelIndex, qmodelindex)
+
+SimpleGetter(QModelIndex, qmodelindex, column, int)
```

Comment:
I like!!

If we can't auto-generate this stuff, let's at least use some macros that make things easier to manage.

Be careful though, some macros features are compiler-dependent. We compile with clang/gcc/vc++.